pkg/utils/utils.go

```diff
@@ -0,0 +1,16 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+)
+
+// GetSocket returns socket string in the correct format based on address family
+func GetSocket(hostIP string, hostPort int) string {
+	socket := fmt.Sprintf("%s:%d", hostIP, hostPort)
+	ipAddr := net.ParseIP(hostIP)
+	if ipAddr.To4() == nil {
+		socket = fmt.Sprintf("[%s]:%d", hostIP, hostPort)
```

Comment:
(Minor) The only minor issue with this is that if it is any non IPv4 address, such as "1.2.3" or "example.com", the socket string will have square brackets.  
  
Ideally, the preferred behavior is if it's an IPv6 address, put square brackets, otherwise, leave it as-is.  That way, the error message for "1.2.3" with port 2055 won't be that \[1.2.3\]:2055 failed.  Unfortunately, there is no function in IP package to check for IPv6, so you have to resort to something like it has a colon character (which I know some people don't like).  Nevertheless, it's minor so I'm okay going with this solution.

Reply:
there is `To16` in net package but it really doesn't differentiate between IPv4 and IPv6 I ran into uses using it in the pass here is an example where it shows its unexpected behavior https://go.dev/play/p/VRP6XKdfgJy 
so the reality is this hostIP so it can't be malformed format we will have way many issues outside no if that ever happened, hence I used that route

Reply:
fixed it found away so malformed v4 address won't show up as ipv6

Reply:
I like the solution!